### PR TITLE
Ide support

### PR DESCRIPTION
**Pull request by claymccoy**
This automates the things that I needed to do to get the project compiling and running unit tests inside IntelliJ. There is some basic support for Eclipse too. When the IDE's get more mature Gradle integration this can hopefully be removed.

### CODE REVIEW

File: build.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<project>
+    <target name="compile">
+        <exec executable="./gradlew">
+            <arg value="testClasses"/>
+        </exec>
+    </target>
+</project>
\ No newline at end of file
```
Comment (quidryan) [inline build.xml:10]: Can we can get comment in here for what the purpose of this file is, and that people shouldn't actively use Ant.

**Comment by quidryan** [comment]
What are the steps you take? Open as Gradle project, then ./gradlew idea? I'm not sure I'm doing it in the right order, I don't see any of the workspace settings coming through. Do I need to add it a module or can I open it as a project?

**Comment by claymccoy** [comment]
Don't open as IJ gradle project. That is useless. Just generate the IJ files and open it. Regen as needed.

**Comment by claymccoy** [comment]
@quidryan Alright, things should work better out of the box now. There is a case on a completely new project where the tests don't run the very first time, but there doesn't appear to be anything I can change in the idea generated files to fix that.

**Comment by quidryan** [comment]
I'm merging it, but can you also add some documentation in the "How to build page" on how to setup intellij correctly?

**Comment by claymccoy** [comment]
done

**Comment by quidryan** [comment]
Thanks.

**Pull request merged**